### PR TITLE
✨ Add e2e test spec to upgrade or reinstall a Helm chart on a Cluster

### LINKED ISSUE #172: Add e2e tests to support coverage of Helm install, upgrade, and uninstall
**User Story**

I'd like to add additional test coverage beyond just the CAPI quick start and API version upgrade tests. These tests could validate that a Helm install of a chart is successful and deployed using the Helm Get() client, and that when we upgrade a chart, the revision is incremented or reinstalled if the release name/namespace changes. We could also validate the multi-cluster use case to verify that creating/deleting a new cluster with matching labels triggers the cluster selector.

/kind feature

### PR DESCRIPTION
**Pull request by Jont828**
**What this PR does / why we need it**: Adds an additional spec to update a HelmChartProxy and verify that the revision and values have successfully changed.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #172

### CODE REVIEW

File: test/e2e/helpers.go [hunk 1]
```diff
@@ -176,20 +179,21 @@ func getHelmActionConfigForTests(ctx context.Context, workloadClusterProxy frame
 
 	actionConfig := new(helmAction.Configuration)
 	klog.Info("Initializing action config")
-	err := actionConfig.Init(settings.RESTClientGetter(), "default", "secret", Logf)
+	// TODO: validate that this is the right namespace
```
Comment (mboersma) [inline test/e2e/helpers.go:182]: We should implement or remove this `// TODO:`.

**Review by mboersma — COMMENTED** [review]
lgtm, but for the `// TODO:`.

**Comment by Jont828** [comment]
/hold
While waiting for tests to pass

**Review by mboersma — APPROVED** [review]
/lgtm

**Pull request merged**